src/utils/breakdown_validation.py

```diff
@@ -115,54 +148,224 @@ def greater_than_validation(
+            ]
+            cols += [c for c in check_df.columns if c not in cols] + [f"{key}_diff"]
+
+            # add the mini-dataframe to the dict for unit testing and logger messages
+            check_results_dict[key] = check_df
+        else:
+            check_results_dict[key] = pd.DataFrame()
+
+    # Filter and select so the qa dataframe has only relevant columns and rows
+    qa_df = qa_df.loc[df["reference"].isin(wanted_refs)][cols]
+
+    return check_results_dict, qa_df
+
+
+def log_errors_to_screen(check_results_dict: dict, check_type: str) -> None:
+    """Function to log the errors to the screen.
 
-    if check != 'all':
-        df = pd.concat([validation_df, not_for_validating_df], ignore_index=True)
+    Args:
+        check_results_dict (dict): The dictionary of errors to log.
+
+    Returns:
+        None
+    """
+    for key, value in check_results_dict.items():
+        if not value.empty:
+            BreakdownValidationLogger.error(
+                f"Breakdown validation failed for {key} columns"
+            )
+            BreakdownValidationLogger.error(value)
     else:
-        df = validation_df
+        BreakdownValidationLogger.info(f"All {check_type} breakdown vals are valid.")
+
+
+def run_imputation_breakdown_validation(df: pd.DataFrame, config: dict) -> None:
+    """Function to run the breakdown validation for the imputed data.
 
-    BreakdownValidationLogger.info(f"There are {count} errors with the breakdown values")
+    Args:
+        df (pd.DataFrame): The dataframe to check.
+        config (dict): The config dictionary.
+
+    Raises:
+        ValueError: If any of the breakdown values do not sum to the total.
+
+    Returns:
+        None
+    """
+    to_check_dict = get_equality_dicts(config, "imputation")
+    check_results_dict, qa_df = get_breakdown_errors(df, to_check_dict)
+    # temp output of qa_df for debugging
+    if not qa_df.empty:
+        print(qa_df)
 
+    log_errors_to_screen(check_results_dict, "imputation")
+
+
+def run_construction_breakdown_validation(df: pd.DataFrame, config: dict) -> None:
+    """Function to run the breakdown validation for the constructed data.
+
+    Few errors are expected, so any that exist will be logged to the screen.
+
+    Args:
+        df (pd.DataFrame): The dataframe to check.
+        config (dict): The config dictionary.
+
+    Returns:
+        pd.DataFrame
+    """
+    to_check_dict = get_equality_dicts(config)
+    df = replace_nulls_with_zero(df, to_check_dict)
+
+    rows_to_validate = remove_all_nulls_rows(df, config)
+    msg, count = equal_validation(rows_to_validate, to_check_dict)
+    msg, count = greater_than_validation(rows_to_validate, msg, count)
+
+    BreakdownValidationLogger.info(
+        f"There are {count} errors with the breakdown values"
+    )
     if not msg:
         BreakdownValidationLogger.info("All breakdown values are valid.")
     else:
         raise ValueError(msg)
 
     return df
```

Comment:
I wonder if we comment out the value error (and returning the updated df), and just log the error message to the screen at this point?  Given we still have questions would it be safer for the moment